### PR TITLE
add neptune backend protobuf calls for series values fetching.

### PR DESCRIPTION
**Pull request by szysad**
Additionally:
  - add unit test for new call. Includes some refractoring as new calls are ment to be new implementation of already exisiting calls.
  - add protobuf python compliled code with stubs in neptune.internal.proto with additional unit tests
  - pin types-protobuf in mypy pre-commit hook as a workaround to fresh protobuf-typing bug. More info: https://github.com/python/mypy/issues/16767
  - excluded proto files from flake8 check as they are output of protoc compiler and far from perfect syntax
  - install tensorflow dev dependency only if python version >=3.10
  - skip executing unit tests that require tensorflow if it's not installed. 

## Before submitting checklist

- [ ] <s>Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)</s>
- [ ] <s>Did you **ask the docs owner** to review all the user-facing changes?</s>

### CODE REVIEW

File: tests/unit/neptune/new/internal/backends/test_neptune_backend_mock.py [hunk 1]
```diff
@@ -270,7 +268,13 @@ def test_get_string_series_values(self):
                     ret,
                 )
 
-    def test_get_float_series_values(self):
+    def test_get_string_series_values(self):
```
Comment (szysad) [inline tests/unit/neptune/new/internal/backends/test_neptune_backend_mock.py:271]: rename `test_get_string_series_values ` to `test_get_float_series_values`

File: pyproject.toml [hunk 2]
```diff
@@ -212,6 +213,10 @@ force_grid_wrap = 2
 [tool.flake8]
 max-line-length = 120
 extend-ignore = "E203"
+exclude = [
+    # protobuf compiler output
+    "src/neptune/internal/proto/protobuf_api_series_dto_pb2.py"
```
Comment (Raalsky) [inline pyproject.toml:218]: Does "just" `src/neptune/internal/proto/` works? It's probably more future-proof

File: src/neptune/internal/backends/neptune_backend.py [hunk 3]
```diff
@@ -211,6 +211,12 @@ def get_float_series_attribute(
     ) -> FloatSeriesAttribute:
         pass
 
+    @abc.abstractmethod
+    def get_float_series_values_proto(
```
Comment (Raalsky) [inline src/neptune/internal/backends/neptune_backend.py:215]: I think this is not needed (no point of extending the abstract class/interface) as this is a part of implementation detail what protocol/endpoint to use. Could we just have two private methods as a part of `HostedNeptuneBackend` and keep the interface the same as it was before?

Comment (szysad) [reply src/neptune/internal/backends/neptune_backend.py:215]: Yes we can do that. The only issue with that it requires client config support flag which was out of scope of this PR.

For now I have implemented proto support bloating `NeptuneBackend` method pool in next PR (attached on top of this one) https://github.com/neptune-ai/neptune-client/pull/1619

```
HostedNeptuneBackend now for both fetching string and float series values has methods:
get_*_series_values_json: that works as before with json dto
get_*_series_values_proto: with new implementation using protobuf
get_*_series_values: chooses proto or json implementation based on client config
```

Comment (Raalsky) [reply src/neptune/internal/backends/neptune_backend.py:215]: > This change requires client config proto support flag which was out of scope of this PR

You can just add a private method in `HostedNeptuneBackend` without any calls of it for now. No need to worry about client config in this PR as you already mentioned.

File: pyproject.toml (hunk 2 above)
Comment (szysad) [reply pyproject.toml]: Yep, good catch!

File: src/neptune/internal/backends/neptune_backend.py (hunk 3 above)
Comment (szysad) [reply src/neptune/internal/backends/neptune_backend.py]: At the end I just implmented `*_proto` methods as private methods of `HostedNeptuneBacked` without changing `NeptuneBackend` interface

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 4]
```diff
@@ -981,6 +1006,31 @@ def get_float_series_values(
...
+        limit: int,
+    ) -> FloatSeriesValues:
+        params = {
+            "experimentId": container_id,
+            "attribute": path_to_str(path),
+            "limit": limit,
+            "offset": offset,
+        }
+        try:
+            result = self.leaderboard_client.api.getFloatSeriesValuesProto(**params).response().result
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:1025]: I think this is not going to work with the final version of our Swagger. @asledz what version are we currently using? The one with protobuf-related classes or the clean one?

Comment (szysad) [reply src/neptune/internal/backends/hosted_neptune_backend.py:1025]: Me and @asledz made some tests and verified it works fine with the final swagger version 🚀

**Comment by szysad** [comment]
✅ E2E
https://github.com/neptune-ai/neptune-client/actions/runs/7655899511

**Pull request closed without merging**